### ISSUE TITLE
MSVC /analyze static analyzer

### ISSUE THREAD

**Issue by illume** [pygame/pygame#1328]
https://docs.microsoft.com/en-us/cpp/build/reference/analyze-code-analysis

**Comment by MyreMylar** [comment]
The PR above adds analyze to appveyor. There are some slight annoyances with the analysis data itself - disabling analysis warnings from system headers works, but the disabling itself generates warnings only on appveyor & some warnings are probably just failures of static analysis but overall I think the output is useful and you can search the generated warning error codes to usually get a pretty good Microsoft help page on what the warning is about, and often how to resolve it.

 `/analyze` also, unsurprisingly slows down build times. Build times on appveyor are quite good though at about 1-2 minutes each build now and only 2-3 minutes with analyze so it's probably not that big of a deal. I suspect we could set it up so analyze only runs on a few of the builds if it gets too slow.

If the initial PR gets checked in I can revist this periodically to try and squash more of the warnings.

**Comment by illume** [comment]
This is merged in now. 

We can leave the issue open to address the remaining warnings it reports.

**Comment by illume** [comment]
I think we should perhaps change /analyze to only run on one job.
- 14 mins (with no /analyze)
- 24 mins (with /analyze)

Recently there's been lots of PRs, commits, and such. With that when several PRs are merged it can take an hour or more to get through them. eg. 5*24 == 120 mins.

If we have to pick one, then probably we should use the most recent compiler /analyze? Since it's probably greatly improved over older versions of the compiler. Perhaps a 32bit one, and a 64bit one?

**Comment by MyreMylar** [comment]
Yeah, I think maybe just leave it on one for now and maybe rotate it later if we run out of warnings. I think we can configure that on appveyor pretty easily IIRC. I'll make a PR sticking it on the 3.8 64 bit build first.

**Comment by MyreMylar** [comment]
PR here:

https://github.com/pygame/pygame/pull/1867

**Comment by ankith26** [comment]
Seems like this issue can be closed now, the static analyzer works well on Windows appveyor builds RN
